tm2/pkg/bft/privval/signer/remote/tcp_conn.go

```diff
@@ -0,0 +1,84 @@
+package remote
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"slices"
+	"time"
+
+	"github.com/gnolang/gno/tm2/pkg/crypto/ed25519"
+	p2pconn "github.com/gnolang/gno/tm2/pkg/p2p/conn"
+)
+
+// Errors returned by the TCP connection configuration.
+var (
+	ErrUnauthorizedPubKey = errors.New("unauthorized remote public key")
+	ErrSecretConnFailed   = errors.New("secret connection handshake failed")
+	ErrNilConn            = errors.New("nil connection")
+)
+
+// configureTCPConnection configures the linger and keep alive options for a TCP connection.
+// It also secures the connection and mutually authenticates with the remote peer using the
+// provided localPrivKey and a list of authorizedKeys.
+func ConfigureTCPConnection(
+	conn *net.TCPConn,
+	localPrivKey ed25519.PrivKeyEd25519,
+	authorizedKeys []ed25519.PubKeyEd25519,
```

Comment:
These can be options

Reply:
What do you mean?

Reply:
Just for these 2:
```go
	keepAlivePeriod time.Duration,
	handshakeTimeout time.Duration,
```

We can use `WithKeepAlivePeriod` and `WithHandshakeTimeout`

Reply:
Done https://github.com/gnolang/gno/pull/3554/commits/a92bfad15ac3b81f09a58e05e1f0029abcb5c81d